index.js

```diff
@@ -196,7 +199,7 @@ KafkaLogger.prototype.log = function(level, msg, meta, callback) {
     logMessage.msg = msg;
     logMessage.fields = meta;
 
-    if ((!this.connected || (this.kafkaRestClient && !this.kafkaRestClientConnected))  && Date.now() < this.initTime + 5000) {
+    if ((this.kafkaClient && !this.connected) || (this.kafkaRestClient && !this.kafkaRestClientConnected) && Date.now() < this.initTime + 5000) {
```

Comment:
&& is prior to ||, need extra parentheses
((this.kafkaClient && !this.connected) || (this.kafkaRestClient && !this.kafkaRestClientConnected)) && Date.now() < this.initTime + 5000